File: client/game/gameState.js
/**
 * Represents a game's state.
 * @param {Player} player1
 * @param {Player} player2
 * @param {Target[]} targets
 * @param {Projectile[]} projectile
 */
export class GameState {
    constructor(player1, player2, targets, projectiles) {
        this.player1 = player1;
        this.player2 = player2;
        this.targets = targets;
    }
}

/**
 * Represents a player's state.
 * @param {number} x
 * @param {number} y
 * @param {number} health
 */
export class Player {
    constructor(x, y, health) {
        this.x = x
        this.y = y
        this.health = health;
    }
}


/**
 * Represents a target's state.
 * @param {number} x
 * @param {number} y
 * @param {number} velocity
 * @param {boolean} isEnabled
 * @param {number} diameter
 */
export class Target {
    constructor(x, y, velocity, isEnabled, diameter) {
        this.x = x
        this.y = y
        this.diameter = diameter
        this.velocity = velocity;
        this.isEnabled = isEnabled;
    }
}


/**
 * Represents a target's state.
 * @param {number} x
 * @param {number} y
 * @param {number} velocity
 * @param {boolean} isEnabled
 * @param {number} diameter
 */
export class Projectile {
    constructor(x, y, velocity, isEnabled, diameter) {
        this.x = x
        this.y = y
        this.diameter = diameter
        this.velocity = velocity;
        this.isEnabled = isEnabled;
    }
}

File: server/gamestate.go
package main
import(
	"time"
)

/*
CONSTANTS THAT DEFINE ALL GAME STATES *USE THEM CAREFULLY*
*/

// this is the number of targets structures, not the current number of enabled targets.
const NUM_TARGETS int = 3
const CANVAS_WIDTH = 400;
const CANVAS_HEIGHT = 400;
const PLAYER_MOVE_LENGTH = 50;
const TICK_DURATION = 1 * time.Second

type Gamestate struct {
	player1 Player;
	player2 Player;
	targets []Target;
	projectiles []Projectile;
}

type Player struct {
	x int;
	y int;
	diameter int;
	velocity int;
	idx int;
}

type Target struct {
	x int;
	y int;
	diameter int;
	velocity int;
	isEnabled bool;
}

type Projectile struct {
	x int;
	y int;
	diameter int;
	velocity int;
	isEnabled bool;
	forceMultiplier float64;
}
